File: CS/Entities/Form1.cs
using System;
using System.Linq;
using System.Data.Objects;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace DXSample {
    public partial class Form1 : XtraForm {
        NorthwindEntities northwind;
        public Form1() {
            InitializeComponent();
            northwind = new NorthwindEntities();
            ObjectQuery<Product> productsQuery = northwind.Products.Include("Order_Details.Order");
            gridControl1.DataSource = new BindingSource(productsQuery, "");
        }

        private void OnMasterRowEmpty(object sender, DevExpress.XtraGrid.Views.Grid.MasterRowEmptyEventArgs e) {
            Product p = (Product)gridView1.GetRow(e.RowHandle);
            e.IsEmpty = p.Order_Details.Count == 0;
        }

        private void OnMasterRowGetRelationCount(object sender, DevExpress.XtraGrid.Views.Grid.MasterRowGetRelationCountEventArgs e) {
            e.RelationCount = 1;
        }

        private void OnMasterRowGetRelationName(object sender, DevExpress.XtraGrid.Views.Grid.MasterRowGetRelationNameEventArgs e) {
            e.RelationName = "Orders";
        }

        private void OnMasterRowGetChildList(object sender, DevExpress.XtraGrid.Views.Grid.MasterRowGetChildListEventArgs e) {
            Product p = (Product)gridView1.GetRow(e.RowHandle);
            e.ChildList = p.Order_Details.Select(od => od.Order).Distinct().ToList(); ;
        }
    }
}

File: VB/Entities/Form1.vb
Imports System.Data.Objects
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace DXSample

    Public Partial Class Form1
        Inherits XtraForm

        Private northwind As NorthwindEntities

        Public Sub New()
            InitializeComponent()
            northwind = New NorthwindEntities()
            Dim productsQuery As ObjectQuery(Of Product) = northwind.Products.Include("Order_Details.Order")
            gridControl1.DataSource = New BindingSource(productsQuery, "")
        End Sub

        Private Sub OnMasterRowEmpty(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.MasterRowEmptyEventArgs)
            Dim p As Product = CType(gridView1.GetRow(e.RowHandle), Product)
            e.IsEmpty = p.Order_Details.Count = 0
        End Sub

        Private Sub OnMasterRowGetRelationCount(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.MasterRowGetRelationCountEventArgs)
            e.RelationCount = 1
        End Sub

        Private Sub OnMasterRowGetRelationName(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.MasterRowGetRelationNameEventArgs)
            e.RelationName = "Orders"
        End Sub

        Private Sub OnMasterRowGetChildList(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.MasterRowGetChildListEventArgs)
            Dim p As Product = CType(gridView1.GetRow(e.RowHandle), Product)
            e.ChildList = p.Order_Details.[Select](Function(od) od.Order).Distinct().ToList()
        End Sub
    End Class
End Namespace
